### PR TITLE
Active-Active Database

### PR DESCRIPTION
**Pull request by greg-oc**
Remaining work is to add source_ips examples and fix the update acceptance test

### CODE REVIEW

File: internal/provider/resource_rediscloud_active_active_subscription_database.go [hunk 1]
```diff
@@ -463,76 +472,88 @@ func resourceRedisCloudActiveActiveSubscriptionDatabaseUpdate(ctx context.Contex
...
+		globalAlerts = append(globalAlerts, &databases.UpdateAlert{
 			Name:  redis.String(dbAlert["name"].(string)),
 			Value: redis.Int(dbAlert["value"].(int)),
 		})
 	}
 
+	globalSourceIps := setToStringSlice(d.Get("global_source_ips").(*schema.Set))
+
 	// Make a list of region-specific configurations
+	// TODO: need to set the region not in the override list to the global values and remove the override from the state
```
Comment (bengesoff) [inline internal/provider/resource_rediscloud_active_active_subscription_database.go:488]: Is this just needing more time or is there something blocking this?

Comment (greg-oc) [reply internal/provider/resource_rediscloud_active_active_subscription_database.go:488]: Sorry I should have removed that comment, that should be the current behaviour now

**Review by bengesoff — APPROVED** [review]
LGTM but I'll let @pvegh-oc make the final review ⛴️

File: internal/provider/resource_rediscloud_active_active_subscription_database.go [hunk 2]
```diff
@@ -339,81 +344,85 @@ func resourceRedisCloudActiveActiveSubscriptionDatabaseRead(ctx context.Context,
...
-	for _, region_db := range db.CrdbDatabases {
+	var regionDbConfigs []map[string]interface{}
+	publicEndpointConfig := make(map[string]interface{})
+	privateEndpointConfig := make(map[string]interface{})
+	for _, regionDb := range db.CrdbDatabases {
+		// Set the endpoints for the region
+		publicEndpointConfig[redis.StringValue(regionDb.Region)] = redis.StringValue(regionDb.PublicEndpoint)
+		privateEndpointConfig[redis.StringValue(regionDb.Region)] = redis.StringValue(regionDb.PrivateEndpoint)
+		// Check if the region is in the state as an override
+		x := getStateOverrideRegion(d, redis.StringValue(regionDb.Region))
```
Comment (pvegh-oc) [inline internal/provider/resource_rediscloud_active_active_subscription_database.go:375]: Can we come up with a more reasonable name on the variable? x is not that descriptive...

Comment (greg-oc) [reply internal/provider/resource_rediscloud_active_active_subscription_database.go:375]: Good spot, definitely, I'll update that now

**Review by pvegh-oc — APPROVED** [review]

**Pull request merged**